Flaky tests on IE11

### Description
`vaadin-text-field` has been having random failures in CI tests often about assertion failure outside test function mainly in IE11 but not every time.

The error may look like this for example:

```
Windows 10 IE 11         Tests failed: Error thrown outside of test function: AssertionError: expected true to be false (http://localhost:8081/components/chai/chai.js:265)
...
Test run ended in failure: Error thrown outside of test function: AssertionError: expected true to be false (http://localhost:8081/components/chai/chai.js:265)
...
Windows 10 IE 11 (273/6/0 error)
...
Error: Error thrown outside of test function: AssertionError: expected true to be false (http://localhost:8081/components/chai/chai.js:265)
```

This seems to be caused by doing `expect()` inside an event listener in a test but not removing the event listener and in some cases the event might be triggered again after the test has finished (causing a failure during an unrelated test or outside test function maybe even after all tests have passed already).

This might not always be a problem but at least the `animationend` event seems to be triggering again after the test sometimes (randomly) in IE11.

I was able to reproduce this locally in a VM with IE11 when running the tests several times.

In most cases where we use event listeners in a test it's just using a sinon spy as an event listener which should be fine (since the spy doesn't directly do `expect()` or throw errors).

We should always make sure to remove the event listeners added during a test at least if the event listener can throw an error or contains an assertion like `expect()`.

### How to fix

Currently there are 4 tests that do `expect()` directly in the event listener. Two with `change` event and two with `animationend` event. These tests should be update to use a helper like this https://github.com/vaadin/vaadin-grid/blob/master/test/helpers.html#L94 to wrap the listener so that it removes itself after being executed once.

`vaadin-text-field` doesn't yet have a helpers file so that should probably be created containing only this one helper for now.

--- a/test/.eslintrc.json
+++ b/test/.eslintrc.json
@@ -9,8 +9,8 @@
     "expect": false,
     "gemini": false,
     "sinon": false,
-    "HTMLImports": false,
     "MockInteractions": false,
-    "animationFrameFlush": false
+    "animationFrameFlush": false,
+    "listenOnce": false
   }
 }

--- a/test/custom-input.html
+++ b/test/custom-input.html
@@ -10,6 +10,7 @@
   <link rel="import" href="../vaadin-text-field.html">
   <link rel="import" href="../../iron-form/iron-form.html">
 
+  <link rel="import" href="helpers.html">
 </head>
 
 <body>
@@ -48,7 +49,7 @@
 
       it('should dispatch change event on slotted input change', done => {
         const changeEvent = new Event('change');
-        textField.addEventListener('change', e => {
+        listenOnce(textField, 'change', e => {
           expect(e.detail.sourceEvent).to.equal(changeEvent);
           done();
         });

--- a/test/helpers.html
+++ b/test/helpers.html
@@ -1,0 +1,9 @@
+<script>
+  window.listenOnce = (element, eventName, callback) => {
+    const listener = e => {
+      element.removeEventListener(eventName, listener);
+      callback(e);
+    };
+    element.addEventListener(eventName, listener);
+  };
+</script>

--- a/test/text-area.html
+++ b/test/text-area.html
@@ -9,6 +9,7 @@
   <link rel="import" href="../../test-fixture/test-fixture.html">
   <link rel="import" href="../vaadin-text-area.html">
 
+  <link rel="import" href="helpers.html">
 </head>
 
 <body>
@@ -164,7 +165,7 @@
           textArea.hidden = true;
           // Three new lines will expand initial height
           textArea.value = '\n\n\n';
-          textArea.addEventListener('animationend', e => {
+          listenOnce(textArea, 'animationend', () => {
             expect(textArea.clientHeight).to.be.above(savedHeight);
             done();
           });
@@ -173,7 +174,7 @@
 
         it('should not update height on custom animation name', done => {
           const spy = sinon.spy(textArea, '_updateHeight');
-          textArea.addEventListener('animationend', e => {
+          listenOnce(textArea, 'animationend', () => {
             expect(spy.called).to.be.false;
             done();
           });

--- a/test/text-field.html
+++ b/test/text-field.html
@@ -10,6 +10,7 @@
   <link rel="import" href="../vaadin-text-field.html">
   <link rel="import" href="../../iron-form/iron-form.html">
 
+  <link rel="import" href="helpers.html">
 </head>
 
 <body>
@@ -387,7 +388,7 @@
         it('should dispatch change event on native input change', done => {
           const changeEvent = new Event('change');
 
-          textField.addEventListener('change', e => {
+          listenOnce(textField, 'change', e => {
             expect(e.detail.sourceEvent).to.equal(changeEvent);
             done();
           });

